pkg/dedup/merge_iter.go

```diff
@@ -49,26 +56,25 @@ func (m *mergedSeries) Iterator(_ chunkenc.Iterator) chunkenc.Iterator {
 }
 
 type quorumValuePicker struct {
-	currentValue int64
+	currentValue float64
 	cnt          int
 }
 
 func NewQuorumValuePicker(v float64) *quorumValuePicker {
 	return &quorumValuePicker{
-		currentValue: int64(v),
+		currentValue: v,
 		cnt:          1,
 	}
 }
 
 // Return true if this is the new majority value.
 func (q *quorumValuePicker) addValue(v float64) bool {
-	iv := int64(v)
-	if q.currentValue == iv {
+	if q.currentValue == v {
```

Comment:
Using ` == ` to compare float values is usually not recommended, but I think it should be fine in this context because the values come from the same source of truth.

Reply:
agreed, the value should be the same for given timestamp